### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by bajtos**
- feat: add `npm run outdated` script to check outdated dependencies in the root project and all
lerna packages.
-  chore: update dependencies to their latest versions (as discovered by the new script)

## Checklist

- [x] `npm test` passes on your machine
- New tests added or existing tests modified to cover all changes
- Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- API Documentation in code was updated
- Documentation in [/docs/site](../tree/master/docs/site) was updated
- Affected artifact templates in `packages/cli` were updated
- Affected example projects in `examples/*` were updated

### CODE REVIEW

File: packages/testlab/package.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
     "@types/express": "^4.11.1",
     "@types/fs-extra": "^5.0.1",
     "@types/shot": "^3.4.0",
-    "@types/sinon": "^4.3.0",
+    "@types/sinon": "^5.0.2",
```
Comment (raymondfeng) [inline packages/testlab/package.json]: This is breaking - https://travis-ci.org/strongloop/loopback-next/jobs/435495036

Comment (marioestradarosa) [reply packages/testlab/package.json]: Interesting, why locally all tests are ok?:

Comment (bajtos) [reply packages/testlab/package.json]: Hmm, maybe I did not correctly reinstall dependencies? IDK.

**Comment by bajtos** [comment]
Fixing the build failure turned out to be more difficult than I anticipated!

Please take a look at the two new commits I have added. GitHub is sorting the commits based on the commit time, but the actual history is ordered as follows:

1. feat: add "npm run outdated" script 
2. fix: clean up dataSource usage
3. feat(testlab): add StubbedInstanceWithSinonAccessor
4. chore: update dependencies to their latest versions

@raymondfeng LGTY now?

#### 1 fix: clean up dataSource usage 1473def. 

This one is not strictly necessary after the second commit changed the way how we use Sinon Stubs, but I think it still worth as a cleanup improving the consistency of our code base and application we are scaffolding.

#### 2. feat(testlab): add StubbedInstanceWithSinonAccessor 55f2868

Add two new sinon-related helpers to our testlab:

- An interface `StubbedInstanceWithSinonAccessor` describing a stubbed
  instance that provides access to Sinon API for individual methos
  via a `stubs` property.

- A wrapper function `createStubInstance` that calls
  `sinon.createStubInstance` under the hood and returns
  `StubbedInstanceWithSinonAccessor`.

These two new helpers allow unit test to
 - use Sinon in a way that's easy to understand for TypeScript,
 - avoid unnecessary type casts and
 - stay compatible with the upcoming versions of sinon typings.

**Comment by bajtos** [comment]
(rebased on top of the latest master)

**Comment by raymondfeng** [comment]
@bajtos I still see CI failures:
```
packages/repository/test/unit/repositories/relation.factory.unit.ts(101,5): error TS2322: Type 'SinonStubbedInstance<CustomerRepository>' is not assignable to type 'CustomerRepository'.
  Property 'setupPersistedModel' is missing in type 'SinonStubbedInstance<CustomerRepository>'.
```

**Comment by bajtos** [comment]
> ```
> packages/repository/test/unit/repositories/relation.factory.unit.ts(101,5): error TS2322: Type 'SinonStubbedInstance<CustomerRepository>' is not assignable to type 'CustomerRepository'.
>   Property 'setupPersistedModel' is missing in type 'SinonStubbedInstance<CustomerRepository>'.
> ```

That was most likely introduced by rebase, I'll take a look.

**Comment by bajtos** [comment]
Fixed.

@raymondfeng LGTY now?

**Pull request merged**